covert/blockstream.py

```diff
@@ -117,9 +115,11 @@ def decrypt_blocks(self):
           block = memoryview(fut.result())
           nextlen = int.from_bytes(block[-3:], "little")
           self.blkhash = sha512(self.blkhash + self.ciphertext[p + elen - 16:p + elen]).digest()
-          self.filepos += len(block) + 16
           yield block[:-3]
           if len(self.q) < self.workers:
+            # Roll back pos to last decoded block end whenever the queue is emptied
+            if not self.q: self.pos = p + elen
```

Comment:
This seems odd. Shouldn't self.pos be updated when new blocks are queued with self._add_to_queue().

Also, below there is a line "pos = self._add_to_queue(p, extlen)" which creates a variable pos that is not used anywhere. It yells to me that it should be assigned to self.pos.

Reply:
The original code was very messy and was largely restructured a while back. As you can see, some problems remain. Your review is quite correct, the missing `self.` was the original problem, so this addition was not actually needed.